R/methods-CogapsResult.R

```diff
@@ -426,20 +426,21 @@ function(object, threshold, lp, axis)
     )
     markerRanks <- apply(markerScores, 2, rank)
     colnames(markerScores) <- colnames(markerRanks) <- colnames(normedMatrix)
+
     ## keep only a subset of markers for each pattern depending on the type of threshold
-    if (threshold == "cut") # all markers which achieve minimum rank
-    {
+    if (threshold == "cut") {#all markers which achieve minimum rank
+
+      #define a ranking function
       simplicityGENES <- function(As, Ps) {
         # rescale p's to have max 1
-        pscale <- apply(Ps,1,max)
-        
+        pscale <- apply(Ps, 2, max)
+
         # rescale A in accordance with p's having max 1
         As <- sweep(As, 2, pscale, FUN="*")
-        
+
         # find the A with the highest magnitude
         Arowmax <- t(apply(As, 1, function(x) x/max(x)))
-        pmax <- apply(As, 1, max)
-        
+
```

Comment:
Thanks, to the point! The disagreement between the dimensions and hence inability to do the multiplication on the fly also bothered me. Maybe keeping the A and P discordant in the output was part of the masterplan, or may be it was a bug, but brutally changing output to agreeable dimensions looked too brave as it could break something downstream. So I started out with [tests/testthat/test_output_across_modes.R](https://github.com/FertigLab/CoGAPS/pull/103/files/186dce080fc597867c9adf0f05a8079843fbe344#diff-58622606deca119f6d5b90ad58782f3623e2481d60607176a8473b3e258c3496) to confirm that outputs disagree for all the modes, and confirmed with @jeanettejohnson that we historically let the user orient the matrices.

Reply:
ok, so the $P$ matrix here would be $n\times k$ and this all does make sense?

Reply:
yes, to some historic reason (GIST data test run shown with 7 patterns), the output is 
```
> dim(res@featureLoadings)
[1] 1363    7
> dim(res@sampleFactors)
[1] 9 7
```